### PR TITLE
Custom Cassandra YAML and JVM options.

### PR DESCRIPTION
**Pull request by phrynchanka**
## What changes are proposed in this PR? 
Resolves  [DCOS-59496](https://jira.mesosphere.com/browse/DCOS-59496)
Resolves  [DCOS-59495](https://jira.mesosphere.com/browse/DCOS-59495) 

This PR Implement adding:

- Base64 encoded Cassandra properties appended to cassandra.yaml.
- Base64 encoded JVM Options appended to jvm.options.

Implementation steps : 
1. Added `CUSTOM_CASSANDRA_YAML_BASE64` and `CUSTOM_JVM_OPTIONS_BASE64` properties to `params.yml`
2. Modified `jvm-options.yaml` and `cassandra-yaml.yaml`. 
3. Used [b64dec](http://masterminds.github.io/sprig/) template function to decode base64 property.
## How were these changes tested?
Manually tested  
1. Install operator 
2. Look into pod `cassandra-node-0` from k8s ui. On `exec` console `cat` the content of `etc/cassadra/cassanda.yaml` and `etc/cassadra/jvm.options` and see that properties were decoded and propagated.

### CODE REVIEW

File: operator/templates/cassandra-yaml.yaml [hunk 1]
```diff
@@ -1351,3 +1351,7 @@ data:
     {{ if .Params.OTC_BACKLOG_EXPIRATION_INTERVAL_MS }}
     otc_backlog_expiration_interval_ms: {{ .Params.OTC_BACKLOG_EXPIRATION_INTERVAL_MS }}
     {{ end }}
+
+    {{ if .Params.CUSTOM_CASSANDRA_YAML_BASE64 }}
+        {{ .Params.CUSTOM_CASSANDRA_YAML_BASE64 | b64dec }}
```
Comment (samvantran) [inline operator/templates/cassandra-yaml.yaml]: nit: spacing is off compared to other if blocks

Comment (phrynchanka) [reply operator/templates/cassandra-yaml.yaml]: thanks

**Comment by phrynchanka** [comment]
> Very nice @phrynchanka ! Only one minor nit but can we also add a test? Want to make sure this feature continues to works as we make future updates.

@samvantran @mpereira 
Thanks! Ok, I'm starting test implementation.

File: operator/params.yaml [hunk 2]
```diff
@@ -766,3 +766,11 @@ parameters:
   - name: JVM_OPT_G1R_SET_UPDATING_PAUSE_TIME_PERCENT
     description: "Have the JVM do less remembered set work during STW, instead preferring concurrent GC. Reduces p99.9 latency."
     default: ""
+
+  - name: CUSTOM_CASSANDRA_YAML_BASE64
+    description: "Base64 encoded Cassandra properties appended to cassandra.yaml."
```
Comment (mpereira) [inline operator/params.yaml]:
```suggestion
    description: "Base64-encoded Cassandra properties appended to cassandra.yaml."
```

File: operator/params.yaml [hunk 3]
```diff
@@ -766,3 +766,11 @@ parameters:
...
+    default: ""
+
+  - name: CUSTOM_JVM_OPTIONS_BASE64
+    description: "Base64 encoded JVM Options appended to jvm.options."
```
Comment (mpereira) [inline operator/params.yaml]:
```suggestion
    description: "Base64-encoded JVM options appended to jvm.options."
```

File: tests/suites/sanity_test.go [hunk 4]
```diff
@@ -85,6 +86,32 @@ var _ = Describe(TestName, func() {
 		Expect(configuration[parameter]).To(Equal(desiredValue))
 	})
 
+	It("Customize cassandra.yaml", func() {
```
Comment (mpereira) [inline tests/suites/sanity_test.go]:
```suggestion
	It("Sets Cassandra configuration through custom YAML", func() {
```

File: tests/suites/sanity_test.go [hunk 5]
```diff
@@ -85,6 +86,32 @@ var _ = Describe(TestName, func() {
...
+		parameter := "otc_backlog_expiration_interval_ms"
+		initialValue := "200"
+		desiredValue := "300"
+		desiredValueBase64 := base64.StdEncoding.EncodeToString([]byte("otc_backlog_expiration_interval_ms: 300"))
```
Comment (mpereira) [inline tests/suites/sanity_test.go]: Let's build the string based on `parameter` and `desiredValue`.

Comment (mpereira) [inline tests/suites/sanity_test.go]:
```suggestion
		desiredEncodedProperties := base64.StdEncoding.EncodeToString([]byte("otc_backlog_expiration_interval_ms: 300"))
```

**Comment by phrynchanka** [comment]
Thanks @mpereira !
> Do you intend on adding a test for JVM options as well?

Yep.  Now working on it.

**Comment by phrynchanka** [comment]
Test for jvm options done.

File: tests/suites/sanity_test.go [hunk 6]
```diff
@@ -12,10 +13,10 @@ import (
...
 	// log "github.com/sirupsen/logrus"
 
-	cassandra "github.com/mesosphere/kudo-cassandra-operator/tests/utils/cassandra"
-	k8s "github.com/mesosphere/kudo-cassandra-operator/tests/utils/k8s"
-	kubectl "github.com/mesosphere/kudo-cassandra-operator/tests/utils/kubectl"
-	kudo "github.com/mesosphere/kudo-cassandra-operator/tests/utils/kudo"
+	"github.com/mesosphere/kudo-cassandra-operator/tests/utils/cassandra"
+	"github.com/mesosphere/kudo-cassandra-operator/tests/utils/k8s"
+	"github.com/mesosphere/kudo-cassandra-operator/tests/utils/kubectl"
+	"github.com/mesosphere/kudo-cassandra-operator/tests/utils/kudo"
```
Comment (mpereira) [inline tests/suites/sanity_test.go]: I'm curious about this change.

File: tests/utils/cassandra/cassandra.go [hunk 7]
```diff
@@ -7,17 +7,31 @@ import (
...
+	return getConfigurationFromNodeLogs(namespaceName, instanceName, "org.apache.cassandra.config.Config - Node configuration:\\[(.+)\\]", ";")
+}
+
+func ClusterJvmConfiguration(
+	namespaceName string, instanceName string,
+) (map[string]string, error) {
+	return getConfigurationFromNodeLogs(namespaceName, instanceName, "o.a.c.service.CassandraDaemon - JVM Arguments: \\[(.+)\\]", ",")
+}
+
+func getConfigurationFromNodeLogs(
```
Comment (mpereira) [inline tests/utils/cassandra/cassandra.go]: Nice approach.

**Pull request merged**